src/std/misc/string.ss

```diff
@@ -215,3 +219,24 @@
 	(string-set! str i (random-word-char)))
       str)
     ""))
+
+;; When str receives a single argument, the value is turned into a string.
+;; When it receives several arguments, which all must be strings, the
+;; strings are concatenated into one. When called without an argument an
+;; empty string is returned.
+;;
+;; Examples:
+;;  (str 2.0)               => "2.0"
+;;  (str "hello" ", world") => "hello, world"
+(def str
+  (case-lambda
+    (() "")
+    ((v) (let (s (format (if (representable? v) "~r" "~a") v))
+           (if (and (number? v) (string-suffix? "." s))
+             (string-append s "0")
+             s)))
```

Comment:
this will break for flonums in the `[0.0, 1.0)` range, as they will be printed with a leading dot.

Reply:
It is better to just `(format "~f" n)` for flonums, it will do the right thing.

Reply:
Also if it is a string already, just return it.